homeassistant/components/mqtt/client.py

```diff
@@ -464,16 +464,34 @@ async def _async_unsubscribe(self, topic: str) -> None:
             _raise_on_error(result)
         await self._wait_for_mid(mid)
 
-    async def _async_perform_subscription(self, topic: str, qos: int) -> None:
-        """Perform a paho-mqtt subscription."""
+    async def _async_perform_subscriptions(
+        self, subscriptions: Iterable[tuple[str, int]]
+    ) -> None:
+        """Perform a paho-mqtt subscriptions."""
+
+        def _log_and_subscribe() -> list[tuple[int, int]]:
+            subscribe_list = []
+            for topic, qos in subscriptions:
+                result, mid = self._mqttc.subscribe(topic, qos)
+                subscribe_list.append((result, mid))
+                _LOGGER.debug("Subscribing to %s, mid: %s", topic, mid)
+            return subscribe_list
```

Comment:
Considering how this is used, maybe named the return parameter `result_list`, `subscribe_result_list` or something?